### ISSUE TITLE
Pod creation fails due to: chain 'KUBE-MARK-MASQ' does not exist

### ISSUE THREAD

**Issue by aep** [cilium/cilium#30666]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

with kubernetes v1.28.2
created with ` kubeadm init  --skip-phases=addon/kube-proxy`

cillium will install just fine

```
ℹ️  Using Cilium version 1.15.0
🔮 Auto-detected cluster name: kubernetes
🔮 Auto-detected kube-proxy has not been installed
ℹ️  Cilium will fully replace all functionalities of kube-proxy
```

however, during `cilium connectivity test`
it still tries to use the chains created by kubeproxy, which of course now dont exist.

```
  Warning  FailedCreatePodSandBox  8s (x4 over 48s)  kubelet  (combined from similar events): Failed to create pod sandbox: rpc error: code = Unknown desc = failed to add hostport mapping for sandbox k8s_echo-same-node-c74f867d5-9777z_cilium-test_8a6abf6c-dadb-4107-b1da-01fbcf921038_0(6609ff9b9559da5bc5b42be1ebcc4aba4d037ac8816076b019017eb7f344a69c): failed to execute iptables-restore: exit status 2 (iptables-restore v1.8.9 (nf_tables): Chain 'KUBE-MARK-MASQ' does not exist
```

### Cilium Version

cilium-cli: v0.15.22 compiled with go1.21.6 on linux/amd64
cilium image (default): v1.15.0
cilium image (stable): v1.14.6
cilium image (running): 1.15.0

### Kernel Version

6.1.0-17-amd64

### Kubernetes Version

v1.28.2

### Sysdump

[cilium-sysdump-20240207-174656.zip](https://github.com/cilium/cilium/files/14199665/cilium-sysdump-20240207-174656.zip)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aanm** [comment]
@aojea Can you take a look at this one? I believe it might be related with https://github.com/cri-o/cri-o/pull/6772 🤔

**Comment by aep** [comment]
can no longer reproduce this

**Comment by aojea** [comment]
Sorry @aanm I totally missed this, feel free to ping me in slack next time if you see I don't reply is most probable I missed it

This seems more related to https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3178-iptables-cleanup#remove-kube-mark-masq-from-kube-proxy-and-let-kubelet-own-it

fixed in 1.28 https://github.com/kubernetes/kubernetes/issues/82125
